aten/src/ATen/native/mps/operations/BinaryKernel.mm

```diff
@@ -1,10 +1,387 @@
+                                                          getMetalScalarType(otherTensor),
+                                                          getMetalScalarType(common_dtype),
+                                                          kernel,
+                                                          op,
+                                                          kernel_operator,
+                                                          type);
+      [computeEncoder setComputePipelineState:binaryPSO];
+      [computeEncoder setBuffer:inputBuffer  offset:inputTensorStorage atIndex:0];
+      [computeEncoder setBuffer:otherBuffer  offset:otherTensorStorage atIndex:1];
+      [computeEncoder setBuffer:outputBuffer offset:outputTensor.storage_offset() * outputTensor.element_size() atIndex:2];
+      if (!allContiguous) {
+        [computeEncoder setBytes:iterShapeData.data() length:sizeof(uint32_t) * iterShape.size() atIndex:3];
+        [computeEncoder setBytes:&nDim length:sizeof(uint32_t) atIndex:4];
+        [computeEncoder setBytes:strides.data() length:sizeof(uint32_t) * nDim * nOffsets atIndex:5];
+      }
+
+      NSUInteger tgSize = binaryPSO.maxTotalThreadsPerThreadgroup;
+      if (tgSize > numThreads) {
+          tgSize = numThreads;
+      }
+
+      MTLSize threadGroupSize = MTLSizeMake(tgSize, 1, 1);
+      [computeEncoder dispatchThreads: gridSize
+                threadsPerThreadgroup: threadGroupSize];
+    }
+  });
+}
+
+static
+void dispatch_binary_kernel_mps(const Tensor& self, const Tensor& other, const Tensor& output, const std::string& op, const std::string& kernel_operator) {
+  TensorIterator iter;
+  if (op == "le" || op == "gt" || op == "ge") {
+    iter = TensorIterator::comparison_op(const_cast<Tensor&>(output), self, other);
+  } else {
+    iter = TensorIterator::borrowing_binary_op(output, self, other);
+  }
+
+  dispatch_binary_kernel_mps_(iter, op, kernel_operator);
+}
+
+bool getBinaryKernelOperator(const std::string& op_name, std::pair<std::string, std::string>& kernel_operator) {
+  static std::unordered_map<std::string, std::pair<std::string, std::string>> opToKernelOperator = {
+    {"multiplication",        {"mul", "*" }},
+    {"div_out_mps:",          {"div", "/" }},
+    {"add_out_mps:",          {"add", "+" }},
+    {"sub_out_mps:",          {"sub", "-" }},
+    {"lessThan",              {"le",  "<" }},
+    {"greaterThan",           {"gt",  ">" }},
+    {"greaterThanOrEqualTo",  {"ge",  ">="}},
+  };
+
+  auto it = opToKernelOperator.find(op_name);
+  if (it == opToKernelOperator.end()) {
+    return false;
+  }
+
+  kernel_operator = it->second;
+  return true;
+}
+
+bool dispatchNativeBinaryKernel(const Tensor& self,
+                          const Tensor& other,
+                          const Tensor& output,
+                          const Scalar& alpha,
+                          const std::string& op_name) {
+  if (alpha.toFloat() == 1.0   &&
+     (!self.is_contiguous()    ||
```

Comment:
Since we have the check `tensor.is_contiguous() || tensor.storage_offset()` in various places, lets remember to refactor this as a function in OperartionUtils.mm to reuse. We can do this in a separate PR,